src/gravel/api/user.py

```diff
@@ -0,0 +1,48 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+from logging import Logger
+from fastapi.logger import logger as fastapi_logger
+from fastapi.routing import APIRouter
+from fastapi import Request
+from typing import List
+
+from gravel.controllers.user import UserModel
+
+
+logger: Logger = fastapi_logger
+
+router: APIRouter = APIRouter(
+    prefix="/user",
+    tags=["user"]
+)
+
+
+@router.get("/", name="Get list of users", response_model=List[UserModel])
+async def enumerate_users(request: Request):
+    return await request.app.state.gstate.user_mgr.enumerate()
+
+
+@router.put("/", name="Create a new user")
+async def create_user(user: UserModel, request: Request):
+    return await request.app.state.gstate.user_mgr.add(user)
+
+
+@router.get("/{username}", name="Get a user by name")
+async def get_user(username: str, request: Request):
+    return await request.app.state.gstate.user_mgr.get(username)
```

Comment:
This might be returning "None", which is something we probably want to check and convey something more descriptive to the user, even if by raising an exception with a status code.